snuba/datasets/entities/events.py

```diff
@@ -73,16 +115,85 @@ def __init__(self, custom_mappers: Optional[TranslationMappers] = None) -> None:
         schema = storage.get_table_writer().get_schema()
         columns = schema.get_columns()
 
+        events_pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SelectedStorageQueryPlanBuilder(
+                selector=EventsQueryStorageSelector(
+                    mappers=event_translator
+                    if custom_mappers is None
+                    else event_translator.concat(custom_mappers)
+                )
+            ),
+        )
+
+        errors_pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SelectedStorageQueryPlanBuilder(
+                selector=ErrorsQueryStorageSelector(
+                    mappers=errors_translators
+                    if custom_mappers is None
+                    else errors_translators.concat(custom_mappers)
+                )
+            ),
+        )
+
+        def selector_func(_query: Query) -> Tuple[str, List[str]]:
+            if random.random() < float(state.get_config("errors_query_percentage", 0)):
+                return "events", ["errors"]
+
+            return "events", []
+
+        def callback_func(
+            query: Query, referrer: str, results: List[Result[QueryResult]]
+        ) -> None:
+            primary_result = results.pop(0)
+            primary_result_data = primary_result.result.result["data"]
+
+            for result in results:
+                result_data = result.result.result["data"]
+
+                if result_data == primary_result_data:
+                    metrics.increment("query_result", tags={"match": "true"})
+                else:
+                    metrics.increment("query_result", tags={"match": "false"})
+
+                    if len(result_data) != len(primary_result_data):
+                        sentry_sdk.capture_message(
+                            "Non matching Events result - different length",
+                            level="warning",
+                            tags={
+                                "referrer": referrer,
+                                "query": query.get_body(),
```

Comment:
I think you should use extras here as well.

Reply:
Will do